tests/ImageSharp.Tests/Processing/Processors/ColorMatrix/FilterTest.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Numerics;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using SixLabors.ImageSharp.Tests.TestUtilities.ImageComparison;
+
+using SixLabors.Primitives;
+using Xunit;
+
+namespace SixLabors.ImageSharp.Tests.Processing.Processors.Filters
+{
+    public class FilterTest : FileTestBase
+    {
+        [Theory]
+        [WithFileCollection(nameof(DefaultFiles), DefaultPixelType)]
+        public void ImageShouldApplyFilter<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (Image<TPixel> image = provider.GetImage())
+            {
+                Matrix4x4 brightness = MatrixFilters.CreateBrightnessFilter(0.9F);
+                Matrix4x4 contrast = MatrixFilters.CreateContrastFilter(1.2F);
+                Matrix4x4 saturation = MatrixFilters.CreateSaturateFilter(1.5F);
+                image.Mutate(x => x.Filter(brightness * contrast * saturation));
+                image.DebugSave(provider);
```

Comment:
If the color filters are correct (which I assume is true), it's safe to add reference images + a `CompareToReferenceOutput()` call here.

Reply:
They're all definitely accurate. I just wanted to wait til We have this in before updating/adding test images.